javascript/javascript3/week3/script.js

```diff
@@ -0,0 +1,240 @@
+    fetch(`https://api.github.com/search/repositories?q=user:${username}`).then((msg)=>{
+      return msg.json();
+    }).then((msg)=>{
+      resolve(msg);
+    }).catch((e)=>{
+      reject(e);
+    });
+  });
+  
+  //Fullname, url and owner
+  const userDiv = document.createElement("div");
+  const repOwner = document.createElement("h2");
+  const br = document.createElement("br");
+  repOwner.innerHTML = "Github User: " + username;
+  userDiv.appendChild(repOwner);
+  classmates.appendChild(userDiv);
+  userDiv.classList.add("user-div");
+  
+  getData.then((data)=>{
+    data.items.forEach((rep)=>{
+      const repDiv = document.createElement("div");
+      userDiv.appendChild(repDiv);
+
+      const repName = document.createElement("p");
+      repName.innerHTML = rep.full_name;
+      repDiv.appendChild(repName);
+
+      const repUrl = document.createElement("a");
+      repUrl.href = rep.url;
+      repUrl.innerHTML = "Link";
+      repDiv.appendChild(repUrl);
+
+    })
+  });
+}
+
+renderRepositories("");
+
+//Shopping cart
+const productsDiv = document.getElementById("products");
+const productsPrice = document.getElementById("price");
+const productsUsername = document.getElementById("username");
+
+const APIKEY = "9f2a80d8316b32f573e9";
+
+class Product {
+  constructor(name, price) {
+    this.name = name;
+    this.price = price;
+  }
+
+  convertToCurrency(currency){
+    return new Promise((resolve,reject)=>{
+        fetch(`https://free.currconv.com/api/v7/convert?q=DKK_${currency}&compact=ultra&apiKey=${APIKEY}`).then((msg)=>{
+          return msg.json();  
+        }).then((msg)=>{
+            resolve(msg[Object.keys(msg)[0]] * this.price);
+        }).catch((e)=>{
+            reject(e);
+        });
+    });
+  }
+
+}
+
+class ShoppingCart {
+  constructor() {
+    this.products = [];
+  }
+
+  addProduct(product) {
+    this.products.push(product);
+  }
+
+  removeProduct(product) {
+    let counter = 0;
+    let foundProduct = false;
+
+    this.products.forEach((p)=>{
+        if(p==product){
+            foundProduct = true;
+            return;
+        }
+        counter++;
+    });
+    if(foundProduct){
+        this.products.splice(counter,1);
+    }
+  }
+
+  searchProduct(productName) {
+    let foundProducts = [];
+    this.products.forEach((p)=>{
+        if(p.name == productName){
+            foundProducts.push(p);
+        }
+    });
+    return foundProducts;
+  }
+
+  getTotal() {
+    let price = 0;
+    this.products.forEach((p)=>{
+        price += p.price;
+    });
+    return price;
+  }
+
+  renderProducts() {
+    this.products.forEach((p)=>{
+        const prod = document.createElement("p");
+        prod.innerHTML = p.name + ": " + p.price + " kr";
+        productsDiv.appendChild(prod);
+    });
+    
+  }
+
+  getUser(user) {
+    return new Promise((resolve,reject) =>{
+        fetch(`https://jsonplaceholder.typicode.com/users/${user}`).then((msg)=>{
+            resolve(msg.json());
```

Comment:
Here you resolve this Promise with another Promise. It might be more correct to add another `.then` step to grab the JSON and then call resolve with the final data.